src/SFML/Window/Wayland/WindowImplWayland.cpp

```diff
@@ -0,0 +1,367 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2021 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Window/Wayland/WindowImplWayland.hpp>
+#include <SFML/System/Utf.hpp>
+#include <SFML/System/Err.hpp>
+#include <SFML/System/Mutex.hpp>
+#include <SFML/System/Lock.hpp>
+#include <SFML/System/Sleep.hpp>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <libgen.h>
+#include <fcntl.h>
+#include <algorithm>
+#include <vector>
+#include <string>
+#include <cstring>
+#include <wayland-client-protocol.h>
+#include <xdg-shell-client-protocol.h>
+#include <SFML/Window/EglContext.hpp>
+typedef sf::priv::EglContext ContextType;
+
+
+////////////////////////////////////////////////////////////
+// Private data
+////////////////////////////////////////////////////////////
+namespace
+{
+
+}
+
+namespace sf
+{
+namespace priv
+{
+
+
+
+void WindowImplWayland::registry_handle_global(
+    void * data,
+    struct wl_registry *registry,
+    uint32_t name,
+    char const *interface,
+    uint32_t version
+) {
+    WindowImplWayland *self = static_cast<WindowImplWayland *>(data);
+    if (strcmp(interface, "wl_compositor") == 0)
```

Comment:
```suggestion
    if (std::strcmp(interface, "wl_compositor") == 0)
```

And please update other usages